ledger-core/virtual/execute/shared/saferesponsecounter.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package shared
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+type SafeResponseCounter struct {
+	count  int
+	wakeUp bool
+}
+
+func CounterIncrement(ctx smachine.ExecutionContext, link smachine.SharedDataLink, count int) smachine.StateUpdate {
+	if count == 0 {
+		return smachine.StateUpdate{}
+	}
+
+	accessor := link.PrepareAccess(func(i interface{}) (wakeup bool) {
+		obj, ok := i.(*SafeResponseCounter)
+
+		switch {
+		case !ok:
+			panic(throw.IllegalState())
```

Comment:
panic caused by cast directly will be more informative

Reply:
fixed